python311Packages.html2text: 2020.1.16 -> 2024.2.26

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for python311Packages.html2text is: Turn HTML into equivalent Markdown-structured text

meta.homepage for python311Packages.html2text is: https://github.com/Alir3z4/html2text/




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

closes https://github.com/NixOS/nixpkgs/pull/291360

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 2024.2.26 with grep in /nix/store/pd8cf4gxrr4s3118kqq8818gw2cgyib5-python3.11-html2text-2024.2.26
- found 2024.2.26 in filename of file in /nix/store/pd8cf4gxrr4s3118kqq8818gw2cgyib5-python3.11-html2text-2024.2.26

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
22 total rebuild path(s)

22 package rebuild(s)

First fifty rebuilds by attrpath
calibre
fanficfare
faraday-agent-dispatcher
faraday-cli
kmymoney
mastodon-archive
mealie
moodle-dl
parsedmarc
python311Packages.faraday-plugins
python311Packages.html2text
python311Packages.mailsuite
python311Packages.parsedmarc
python311Packages.trytond
python311Packages.weboob
python311Packages.woob
python312Packages.faraday-plugins
python312Packages.html2text
python312Packages.trytond
rss2email
trytond
unbook
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/pd8cf4gxrr4s3118kqq8818gw2cgyib5-python3.11-html2text-2024.2.26 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A python311Packages.html2text https://github.com/r-ryantm/nixpkgs/archive/b6ce48148f6650e732002b4c08f6508f0e6c788a.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/b6ce48148f6650e732002b4c08f6508f0e6c788a#python311Packages.html2text
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/pd8cf4gxrr4s3118kqq8818gw2cgyib5-python3.11-html2text-2024.2.26
ls -la /nix/store/pd8cf4gxrr4s3118kqq8818gw2cgyib5-python3.11-html2text-2024.2.26/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>rss2email</li>
  </ul>
</details>
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>calibre</li>
    <li>fanficfare</li>
    <li>faraday-agent-dispatcher</li>
    <li>faraday-cli</li>
    <li>kmymoney</li>
    <li>mastodon-archive</li>
    <li>mealie</li>
    <li>moodle-dl</li>
    <li>parsedmarc (python311Packages.parsedmarc)</li>
    <li>python311Packages.faraday-plugins</li>
    <li>python311Packages.html2text</li>
    <li>python311Packages.mailsuite</li>
    <li>trytond (python311Packages.trytond)</li>
    <li>python311Packages.weboob</li>
    <li>python311Packages.woob</li>
    <li>python312Packages.faraday-plugins</li>
    <li>python312Packages.html2text</li>
    <li>python312Packages.trytond</li>
    <li>unbook</li>
  </ul>
</details>

---

###### Maintainer pings

wegank commented:
Breaks `rss2email`.

dotlambda commented on pkgs/development/python-modules/html2text/default.nix:
```diff
@@ -7,15 +7,15 @@
   pname = "html2text";
-  version = "2020.1.16";
+  version = "2024.2.26";
   format = "setuptools";
```
```suggestion
  pyproject = true;
```

dotlambda commented on pkgs/development/python-modules/html2text/default.nix:
```diff
@@ -7,15 +7,15 @@
 
   src = fetchFromGitHub {
     owner = "Alir3z4";
     repo = pname;
```
```suggestion
    repo = "html2text";
```

dotlambda commented on pkgs/development/python-modules/html2text/default.nix:
```diff
@@ -7,15 +7,15 @@
-  version = "2020.1.16";
+  version = "2024.2.26";
   format = "setuptools";
   disabled = pythonOlder "3.5";
```
```suggestion
  disabled = pythonOlder "3.8";
```

dotlambda committed:
python311Packages.html2text: 2020.1.16 -> 2024.2.26

Diff: https://github.com/Alir3z4/html2text/compare/refs/tags/2020.1.16...2024.2.26

Changelog: https://github.com/Alir3z4/html2text/blob/refs/tags/2024.2.26/ChangeLog.rst

dotlambda committed:
rss2email: fix tests with html2text 2024.2.25

wegank commented:
Partially done in https://github.com/NixOS/nixpkgs/commit/563bb5b85e41b085163e0fd4a613092d59a7ddae, please rebase.

wegank commented:
Oh no, why did I close this...
